dotnet/winforms
# fix: Restore missing TF for `System.Windows.Forms.HighDpiMode`

Resolves #2925

`System.Windows.Forms.HighDpiMode` was accidentally placed under `Internal` namespace, which caused a missing type-forward decoration.
This resulted in users receiving the following error when attempting to roll forward from .NET Core 3.x to .NET 5:

```
System.MissingMethodException : Method not found: 'Boolean System.Windows.Forms.Application.SetHighDpiMode(System.Windows.Forms.HighDpiMode)'.
```



## Proposed changes

- Move the type into the correct location and add the missing `TypeForward` decoration.



## Customer Impact

- User can now roll forward from .NET Core 3.x and use `Boolean System.Windows.Forms.Application.SetHighDpiMode(System.Windows.Forms.HighDpiMode)`

## Regression? 

- Yes

## Risk

- Minimal



## Test methodology
* [Use the repro app](https://github.com/RussKie/Test-TypeForwards/commit/c5f76dca248dee7fd69d95fc9b5af0f76c32578e) or generate a new Windows Forms app targeting .NET Core 3.0 (or 3.1) and set it to rollforward:
  ```json
  {
    "sdk": {
      "rollForward": "major"
    }
  }
  ```
* Ensure no .NET Core 3.x runtime is installed. Best to run a VM with only .NET5 Preview1 installed
* Run the app, observe the failure:
  ![image](https://user-images.githubusercontent.com/4403806/75741880-12b02a00-5d60-11ea-9365-2042177f0eff.png)
* Build the current commit with `pack` option:
  ```powershell
  PS> .\build.ps1 -pack
  ```
* Copy and replace `System.Windows.Forms.dll` and `System.Windows.Forms.Primitives.dll`
	- from .\artifacts\packages\Debug\NonShipping\Microsoft.Private.Winforms.5.0.0-dev.nupkg\lib\netcoreapp5.0 --> C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\5.0.0-preview.2.20119.5
	- from .\artifacts\packages\Debug\NonShipping\Microsoft.Private.Winforms.5.0.0-dev.nupkg\ref\netcoreapp5.0 --> C:\Program Files\dotnet\packs\Microsoft.WindowsDesktop.App.Ref\5.0.0-preview.2.20119.5\ref\netcoreapp5.0
* Re-run the app, observe the app start
  ![image](https://user-images.githubusercontent.com/4403806/75742168-eb0d9180-5d60-11ea-9b79-ad4ce33344e1.png)


###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2932)
